extmod/network_wiznet5k.c

```diff
@@ -556,11 +712,28 @@ static mp_uint_t wiznet5k_socket_send(mod_network_socket_obj_t *socket, const by
         wiznet5k_socket_close(socket);
         *_errno = -ret;
         return -1;
+    } else if (ret == SOCK_BUSY) {
+        uint8_t status = getSn_SR(socket->fileno);
+        if (status == SOCK_ESTABLISHED || status == SOCK_CLOSE_WAIT) {
+            *_errno = MP_EAGAIN;
+            return -1;
+        }
     }
     return ret;
 }
 
 static mp_uint_t wiznet5k_socket_recv(mod_network_socket_obj_t *socket, byte *buf, mp_uint_t len, int *_errno) {
+    if (socket->timeout > 0) {
+        mp_uint_t start = mp_hal_ticks_ms();
+        while (getSn_SR(socket->fileno) == SOCK_ESTABLISHED && getSn_RX_RSR(socket->fileno) == 0) {
+            if (mp_hal_ticks_ms() - start > socket->timeout) {
```

Comment:
Because `socket->timeout` is signed `int32_t` this comparison may be signed, but it should be unsigned.  So it needs to be `(uint32_t)(mp_hal_ticks_ms() - start) > (uint32_t)socket->timeout`.